src/validation.ts

```diff
@@ -0,0 +1,123 @@
+import { Client, cacheExchange, fetchExchange } from '@urql/core';
+import {
+  type DscvrFramesRequest,
+  type DscvrValidationResponse,
+  dscvrClientProtocolPrefix,
+  DscvrUntrustedData,
+} from './types';
+import { DEFAULT_DSCVR_API_URL } from './default';
+
+type ValidationType = string | number | bigint | null | undefined;
+interface ValidatedQueryResult {
+  user: {
+    id: string;
+  };
+  content?: {
+    id: string;
+  };
+  state: string | null;
+  inputText: string | null;
+  timestamp: string;
+  url: string;
+  buttonIndex: number;
+}
+
+const validationQuery = `#graphql
+    query validate($message: String!) {
+      unpackFrameMessage(message: $message) {
+        user {
+            id
+        },
+        content {
+            id
+        }
+        buttonIndex,
+        state,
+        url,
+        timestamp,
+        inputText,
+      }
+    }
+`;
```

Comment:
suggestion would be to move this piece of code to `src/queries/validation.query.ts` ?

Reply:
not a blocker